args:
  dtype:
  - Tensor
  - int
  - int
  is_pos:
  - true
  - false
  - false
  name:
  - self
  - source
  - destination
  required:
  - true
  - true
  - true
name: torch.movedim
package: torch
pass_rate: 61.5
rules:
- - cot: 'Based on the given runtime information, the error occurs because the value
      of the `source` parameter is out of the valid range for the given tensor. The
      valid range for the `source` parameter is [-len(self.shape), len(self.shape)-1].
      Therefore, to prevent this error in the future and generate constraints that
      do not trigger it, we need to ensure that the `source` value is always within
      the valid range for the given tensor.


      The constraint for the `source` parameter can be expressed as:'
    length: 1
    target:
      choosen_dtype:
        destination: int
        self: tensor
        source: int
      msg: Dimension out of range (expected to be in range of [-1, 0], but got 3)
      package: torch
    txt: source >= -len(self.shape) and source < len(self.shape) and destination >=
      -len(self.shape) and destination < len(self.shape)
  - f1_score: 72.636815920398
    overall_score: 39.5
    precision: 73.0
    recall: 72.27722772277228
- - cot: default
    length: 1
    target:
      choosen_dtype:
        destination: int
        self: Tensor
        source: int
      msg: negative dimensions are not allowed
      package: torch
    txt: all(i >= 0 for i in self.shape)
  - f1_score: -1
    overall_score: -1
    precision: -1
    recall: -1
- - cot: default
    length: 1
    target:
      choosen_dtype:
        destination: int
        self: Tensor
        source: int
      msg: Too large tensor shape
      package: torch
    txt: self.rank <= 7
  - f1_score: -1
    overall_score: -1
    precision: -1
    recall: -1
